File: src/main/java/com/example/hospital/model/Patient.java
package com.example.hospital.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.File;
import java.sql.Date;
import java.util.List;
import java.util.UUID;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "patients")
public class Patient {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;
    @OneToOne
    @JoinColumn(name = "user_id")
    private User userId;
    private String firstName;
    private String fathersName;
    private String lastName;
    private String dateOfBirth;
    @ManyToOne
    private Address placeOfBirth;
    @ManyToOne
    private Address currentAddress;
    private String imgSrc;
    private String phone;
    private String email;
    private String birthCode;
    private String gender;
    private Date created_at;
    private Date updated_at;
}

File: frontend/src/app/model/patient/patient.ts
export class Patient {
  id:string;
  user:Object;
  firstName:string;
  lastName:string;
  fathersName:string;
  dateOfBirth:string;
  placeOfBirth:string;
  addresses:string[];
  phone:string;
  birthCode:string;
  gender:string;
  created_at:string;
  updated_at:string;

  getPatientNameAndLastNameAndID():string {
    let id = this.id.substring(0, 10);
    let firstName = this.firstName.substring(0, 10);
    let lastName = this.lastName.substring(0, 10);
    return `${id} ${firstName} ${lastName}`;
  }
}



